src/controller/launcher/webots_controller.c

```diff
@@ -169,54 +169,58 @@ static bool get_webots_home() {
 // Gets and stores the path to the latest installed version of Matlab on the system.
 static bool get_matlab_path() {
   struct dirent *directory_entry;  // Pointer for directory entry
+
 #ifdef __APPLE__
   const char *matlab_directory = "/Applications/";
   const char *matlab_version_wc = "MATLAB_R20";
-#else
-  const char *matlab_version_wc = "R20";
-#ifdef _WIN32
+  const char *matlab_exec_suffix = "/bin/matlab";
+#elif _WIN32
   const char *matlab_directory = "C:\\Program Files\\MATLAB\\";
+  const char *matlab_version_wc = "R20";
   const char *matlab_exec_suffix = "\\bin\\matlab.exe";
-#else  // __linux__
+#elif __linux__
   const char *matlab_directory = "/usr/local/MATLAB/";
+  const char *matlab_version_wc = "R20";
   const char *matlab_exec_suffix = "/bin/matlab";
-#endif
+#else
+#error "OS not supported!"
 #endif
 
   DIR *directory = opendir(matlab_directory);
-#ifndef __APPLE__
   if (directory == NULL) {
-    fprintf(stderr, "No installation of MATLAB available.\n");
+#ifdef __APPLE__
+    fprintf(stderr, "Could not open Applications folder to search for matlab installation. Please specify it manually using "
+                    "the option '--matlab-path'.\n");
+#else
+    fprintf(stderr, "No installation of MATLAB found. Please specify it manually using the option '--matlab-path'.\n");
+#endif
     return false;
   }
-#endif
+
   // Get latest available Matlab version
   char *latest_version = NULL;
   while ((directory_entry = readdir(directory)) != NULL) {
     const size_t directory_name_size = strlen(directory_entry->d_name) + 1;
     if (strncmp(matlab_version_wc, directory_entry->d_name, strlen(matlab_version_wc)) == 0) {
-      if (!latest_version)
+      if (!latest_version) {
         latest_version = malloc(directory_name_size);
-      else if (strcmp(latest_version, directory_entry->d_name) < 0)
+        strncpy(latest_version, directory_entry->d_name, directory_name_size);
+      } else if (strcmp(latest_version, directory_entry->d_name) < 0) {
         memset(latest_version, '\0', directory_name_size);
```

Comment:
I believe there is a potential bug here: if `directory_name_size` becomes bigger on a subsequent iteration (which is possible if we have some file or sub-folder with a long name), we may perform a buffer overflow by copying `\0` beyond the boundary of the allocated memory segment, causing potentially a crash. Thus, I believe the `latest_version` buffer should be reallocated to fit the exact size needed for the copy.
```suggestion
        latest_version = realloc(latest_version, directory_name_size);
```

Reply:
agree